docs/build-on-polkadot/xcm/asset-transfer-api/reference.md

```diff
@@ -0,0 +1,280 @@
+    ***Request***
+
+    ```ts
+    --8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/ca-example-request.ts'
+    ```
+
+    ***Response***
+
+    --8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/ca-example-response.md'
+
+
+#### Decode Extrinsic
+
+Decodes the hex of an extrinsic into a string readable format.
+
+```ts
+--8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/de-fn-signature.ts'
+```
+
+??? interface "Request parameters"
+
+    `encodedTransaction` ++"string"++ <span class="required" markdown>++"required"++</span>
+
+    A hex encoded extrinsic.
+
+    ---
+
+    `format` ++"T extends Format"++ <span class="required" markdown>++"required"++</span>
+    
+    Specifies the format for returning a transaction.
+
+    ??? child "Type `Format`"
+
+        ```ts
+        export type Format = 'payload' | 'call' | 'submittable';
+        ```
+
+??? interface "Response parameters"
+
+    ++"string"++
+
+    Decoded extrinsic in string readable format.
+
+??? interface "Example"
+
+    ***Request***
+
+    ```ts
+    --8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/de-example-request.ts'
+    ```
+
+    ***Response***
+
+    --8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/de-example-response.md'
+
+#### Fetch Fee Info
+
+Fetch estimated fee information for an extrinsic.
+
+```ts
+--8<-- 'code/build-on-polkadot/xcm/asset-transfer-api/reference/ffi-fn-signature.ts'
+```
+
+??? interface "Request parameters"
+
+    `tx` ++"ConstructedFormat<T>"++ <span class="required" markdown>++"required"++</span>
+
+    The constructed transaction.
+
+    ??? child "Type `ConstructedFormat<T>`"
+
+        ```ts
+        export type ConstructedFormat<T> = T extends 'payload'
+        ? GenericExtrinsicPayload
+        : T extends 'call'
+        ? `0x${string}`
+        : T extends 'submittable'
+          ? SubmittableExtrinsic<'promise', ISubmittableResult>
+          : never;
+        ```
+
+        The `ConstructedFormat` type is a conditional type that returns a specific type based on the value of the TxResult `format` field.
+
+        - Payload Format - if the format field is set to 'payload', the ConstructedFormat type will return a [GenericExtrinsicPayload](https://github.com/polkadot-js/api/blob/3b7b44f048ff515579dd233ea6964acec39c0589/packages/types/src/extrinsic/ExtrinsicPayload.ts#L48){target=_blank}
+        - Call Format - if the format field is set to 'call', the `ConstructedFormat` type will return a hexadecimal string (`0x${string}`). This is the encoded representation of the extrinsic call
+        - Submittable Format - if the format field is set to `'submittable'`, the `ConstructedFormat` type will return a [`SubmittableExtrinsic`](https://github.com/polkadot-js/api/blob/3b7b44f048ff515579dd233ea6964acec39c0589/packages/api-base/src/types/submittable.ts#L56){target=_blank}. This is a Polkadot.js type that represents a transaction that can be submitted to the blockchain
```

Comment:
Actually, I would extract this `ConstructedFormat<T> to a snippet since it is reused, but just make sure you apply my formatting comments I left on the other instance of it

Reply:
I just noticed the other instance is in another snippet. You can add a snippet to a snippet file if you create a snippet just for the content in the`??? child "Type `ConstructedFormat<T>`"` dropdown